text/text.go

```diff
@@ -0,0 +1,188 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package text
+
+import (
+	"image/color"
+	"math"
+
+	"gonum.org/v1/plot/vg"
+)
+
+// Handler parses, formats and renders text.
+type Handler interface {
+	// Lines splits a given block of text into separate lines.
+	Lines(txt string) []string
+
+	// Box returns the bounding box of the given non-multiline text where:
+	//  - width is the horizontal space from the origin.
+	//  - height is the vertical space above the baseline.
+	//  - depth is the vertical space below the baseline, a positive number.
+	Box(txt string, fnt vg.Font) (width, height, depth vg.Length)
+
+	// Draw renders the given text with the provided style and position
+	// on the canvas.
+	Draw(c vg.Canvas, txt string, sty TextStyle, pt vg.Point)
+}
+
+// XAlignment specifies text alignment in the X direction. Three preset
+// options are available, but an arbitrary alignment
+// can also be specified using XAlignment(desired number).
+type XAlignment float64
+
+const (
+	// XLeft aligns the left edge of the text with the specified location.
+	XLeft XAlignment = 0
+	// XCenter aligns the horizontal center of the text with the specified location.
+	XCenter XAlignment = -0.5
+	// XRight aligns the right edge of the text with the specified location.
+	XRight XAlignment = -1
+)
+
+// YAlignment specifies text alignment in the Y direction. Three preset
+// options are available, but an arbitrary alignment
+// can also be specified using YAlignment(desired number).
+type YAlignment float64
+
+const (
+	// YTop aligns the top of of the text with the specified location.
+	YTop YAlignment = -1
+	// YCenter aligns the vertical center of the text with the specified location.
+	YCenter YAlignment = -0.5
+	// YBottom aligns the bottom of the text with the specified location.
+	YBottom YAlignment = 0
+)
+
+// Position specifies the text position.
+const (
+	PosLeft   = -1
+	PosBottom = -1
+	PosCenter = 0
+	PosTop    = +1
+	PosRight  = +1
+)
+
+// TextStyle describes what text will look like.
+type TextStyle struct {
```

Comment:
Is it worth renaming this to `Style` to avoid stutter?

Reply:
I did consider it.
but the fact it's widely embedded in many structs in `plot` and `plotter` gave me pause.

I haven't checked how `rsc.io/rf` handles that kind of migration.
(a simple `sed` command could handle that though).

let me check.

Reply:
gopls will do that across package boundaries with the label rename.

Reply:
yes. but I'd like the migration for users of gonum/plot (_ie:_ once the new package version has been tagged) to be as easy as `go fix` (or `rf -ex some-script`)
perhaps the tooling we have right now isn't up to the task. yet.

Reply:
A gofmt rewrite should handle that without issue, or sed.

Reply:
done.